core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/SubListSelectorFactory.java

```diff
@@ -93,12 +108,80 @@ private SubListSelector<Solution_> applyMimicRecording(HeuristicConfigPolicy<Sol
         return subListSelector;
     }
 
+    private SubListSelector<Solution_> applyNearbySelection(
+            HeuristicConfigPolicy<Solution_> configPolicy,
+            SelectionCacheType minimumCacheType,
+            SelectionOrder resolvedSelectionOrder,
+            RandomSubListSelector<Solution_> subListSelector) {
+        NearbySelectionConfig nearbySelectionConfig = config.getNearbySelectionConfig();
+        if (nearbySelectionConfig == null) {
+            return subListSelector;
+        }
+
+        randomDistributionNearbyLimitation(nearbySelectionConfig).ifPresent(configPropertyNameAndValue -> {
+            if (config.getMinimumSubListSize() != null && config.getMinimumSubListSize() > 1) {
+                throw new IllegalArgumentException("Using minimumSubListSize (" + config.getMinimumSubListSize()
+                        + ") is not allowed because the nearby selection distribution uses a "
+                        + configPropertyNameAndValue.getKey() + " (" + configPropertyNameAndValue.getValue()
+                        + ") which may limit the ability to select all nearby values."
+                        + " As a consequence, it may be impossible to select a subList with the required minimumSubListSize."
+                        + " Therefore, this combination is prohibited.");
+            }
+        });
+
+        nearbySelectionConfig.validateNearby(minimumCacheType, resolvedSelectionOrder);
+
+        boolean randomSelection = resolvedSelectionOrder.toRandomSelectionBoolean();
+
+        NearbyDistanceMeter<?, ?> nearbyDistanceMeter =
+                configPolicy.getClassInstanceCache().newInstance(nearbySelectionConfig,
+                        "nearbyDistanceMeterClass", nearbySelectionConfig.getNearbyDistanceMeterClass());
+        // TODO Check nearbyDistanceMeterClass.getGenericInterfaces() to confirm generic type S is an entityClass
```

Comment:
TODO

Reply:
Duplicated from existing code. Will be eliminated in a deduplication step or linked to a Jira.